src/Confluent.Kafka/IProducer.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Confluent.Kafka.Impl;
+using Confluent.Kafka.Internal;
+using Confluent.Kafka.Serialization;
+using System.Collections.Concurrent;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Defines a high-level Apache Kafka producer client (without 
+    ///     serialization capability).
+    /// </summary>
+    public interface IProducer : IDisposable
+    {
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Poll_int"]/*' />
+        int Poll(int millisecondsTimeout);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Poll_TimeSpan"]/*' />
+        int Poll(TimeSpan timeout);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Poll"]/*' />
+        [Obsolete("Use an overload of Poll with a finite timeout.", false)]
```

Comment:
I believe there is no controlled way to exit this (pretty sure there isn't an interrupted exception or something that can be caught), so I don't see why it would ever be advisable to use this method, so I don't think we should give people the option. In C# we shouldn't really even have methods with timeouts like this either (without a `CancellationToken`), though I'm fine with doing that. Note also that I expect very few people would be actually using the manual poll option on the .net producer.

What do you think about providing an overload instead that takes a `CancellationToken` and implement it using a loop and calls to poll with a short timeout (checking periodically if the call has been cancelled)?

Reply:
job for a later PR though

Reply:
the `false` in the attribute argument list effectively means 'depreciate'.

Reply:
this was obsoleted in previous versions too so this question can be answered in a subsequent pr.